integration/lerna-publish.test.js

```diff
@@ -56,7 +56,7 @@ describe("lerna publish", () => {
 
   test("exit 0 when no updates", async () => {
     const cwd = await initFixture("normal");
-    const args = ["publish"];
+    const args = ["publish", "--skip-npm"];
```

Comment:
Clearly we need to improve our git setup during integration tests. In fact, I think a bare repository initialized `beforeAll` and then cloned into each `initFixture` temp directory is the way to go. I'll work on implementing that today.

Reply:
To this end, I've added a new `cloneFixture()` helper and an example of usage in https://github.com/lerna/lerna/commit/efbd77beac113e8e6436ecf68494df61ba02b7fc:

```js
const { cwd, repository } = cloneFixture("normal");
const cloneDir = tempy.directory();

// simulate upstream change from another clone
await execa("git", ["clone", repository, cloneDir]);
await fs.outputFile(path.join(cloneDir, "README.md"), "upstream change");
await gitAdd(cloneDir, "-A");
await gitCommit(cloneDir, "upstream change");
await execa("git", ["push", "origin", "master"], { cwd: cloneDir });

try {
  await cliRunner(cwd)("publish");
} catch (err) {
  expect(err.message).toMatch("upstream ahead");
}
```

Reply:
Awesome! I'll add an integration test for this